File: languages/kotlin/src/provider/RangeValueAnalysis.kt
package provider

import box.UserInfo
import model.TableAnalysis

/**
 * Classe para analise de dados, obtem os valores maximo e minimo dos creditos dos usuarios
 * @see model.TableAnalysis
 */
class RangeValueAnalysis: TableAnalysis<Array<Double>> {

    /**
     * Realiza a analise maxima e minima dos dados
     * @param userInfoList Lista de dados a ser analisada
     * @return Valores maximo e minimo dos creditos
     * @see TableAnalysis
     */
    override fun analysis(userInfoList: List<UserInfo>): Array<Double> {

        var min: Double = Double.MAX_VALUE
        var max: Double = Double.MIN_VALUE

        userInfoList.forEach {
            if(it.credit > max) max = it.credit
            if(it.credit < min) min = it.credit
        }

        return doubleArrayOf(min, max).toTypedArray()
    }

}

File: languages/scala/src/provider/RangeValueAnalysis.scala
package provider

import model.{TableAnalysis, UserInfo}

/**
  * Classe para analise de dados, obtem os valores maximo e minimo dos creditos dos usuarios
  * @see model.TableAnalysis
  */
class RangeValueAnalysis extends TableAnalysis[Array[Double]]{

  /**
    * Realiza a analise maxima e minima dos dados
    * @param userInfoList Lista de dados a ser analisada
    * @return Valores maximo e minimo dos creditos
    * @see TableAnalysis
    */
  override def analysis(userInfoList: List[UserInfo]): Array[Double] = {

    var max = Double.MinValue
    var min = Double.MaxValue

    userInfoList.foreach { userInfo =>
      if(userInfo.credit > max) max = userInfo.credit
      if(userInfo.credit < min) min = userInfo.credit
    }

    return Array(min, max)
  }
}
